io-engine/src/replica_backend.rs

```diff
@@ -65,6 +77,51 @@ pub trait ReplicaOps: LogicalVolume {
         -> Result<(), crate::pool_backend::Error>;
 }
 
+#[async_trait::async_trait(?Send)]
+pub trait ReplicaSnapshotOps {
```

Comment:
Do we really need to keep two separate traits `ReplicaSnapshotOps` and `SnapshotOps` when both are in generic backend to be implemented by replica of all backend types? It seems the idea is to separate based on `snapshot creation` vs `clone creation using snapshot`. I'd suggest to either clearly document, or perhaps merge the two traits?

Reply:
`ReplicaSnapshotOps` could be merged into `ReplicaOps`, it's separated only since it's logically a different operation.

`SnapshotOps` should be separate as some of its operations don't make sense for a replica, they're snapshot specific. If we ever add cloning capabilities to replica directly, then we might need some changes there, perhaps.

Reply:
As discussed moved `ReplicaSnapshotOps` into `ReplicaOps`